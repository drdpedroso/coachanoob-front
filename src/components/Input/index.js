import React from 'react'
import PropTypes from 'prop-types';
import { Input } from 'semantic-ui-react'

const BaseInput = (props) => {
    return (
        <Input {...props} onChange={props.onChange} />
    )
};

BaseInput.propTypes = {
    onChange: PropTypes.func
};

export default BaseInput
