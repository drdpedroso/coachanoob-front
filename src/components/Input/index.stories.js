import React from 'react';
import {storiesOf} from '@storybook/react';

import Input from './';
import {Icon} from "semantic-ui-react";

storiesOf('Input', module)
    .add('with placeholder', () => (
        <Input placeholder="Placeholder text"/>
    ))
    .add('with error', () => (
        <Input error value="Value text"/>
    ))
    .add('with fluid', () => (
        <Input fluid value="Value text"/>
    ))
    .add('with icon left', () => (
        <Input iconPosition='left' placeholder='Email'>
            <Icon name='at' />
            <input />
        </Input>
    ))
    .add('with icon right', () => (
        <Input iconPosition='right' placeholder='Email'>
            <Icon name='at' />
            <input />
        </Input>
    ))

