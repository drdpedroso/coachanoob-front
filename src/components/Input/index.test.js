import React from 'react';
import Input from './index'
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<Input/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
