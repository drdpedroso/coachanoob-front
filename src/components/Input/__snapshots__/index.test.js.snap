// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui input"
>
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
